tests/test_pipeline/test_hierarchical_pipeline.py

```diff
@@ -267,3 +280,109 @@ def test_backtest_w_exog(product_level_constant_hierarchical_ts_w_exog, reconcil
+
+
+@pytest.mark.parametrize(
+    "reconciliator",
+    (
+        TopDownReconciliator(target_level="product", source_level="market", period=1, method="AHP"),
+        TopDownReconciliator(target_level="product", source_level="market", period=1, method="PHA"),
+        BottomUpReconciliator(target_level="market", source_level="product"),
+        BottomUpReconciliator(target_level="total", source_level="market"),
+    ),
+)
+def test_forecast_interval(product_level_constant_hierarchical_ts, reconciliator):
+    ts = product_level_constant_hierarchical_ts
+    model = NaiveModel()
+    pipeline = HierarchicalPipeline(reconciliator=reconciliator, model=model, transforms=[], horizon=2)
+
+    pipeline.fit(ts=ts)
+    forecast = pipeline.forecast(prediction_interval=True, n_folds=1)
+    for segment in forecast.segments:
+        target = forecast[:, segment, "target"]
+        np.testing.assert_array_almost_equal(target, forecast[:, segment, "target_0.025"])
+        np.testing.assert_array_almost_equal(target, forecast[:, segment, "target_0.975"])
+
+
+def test_forecast_get_level(product_level_constant_hierarchical_ts):
+    ts = product_level_constant_hierarchical_ts
+    reconciliator = BottomUpReconciliator(target_level="market", source_level="product")
+    model = NaiveModel()
+    pipeline = HierarchicalPipeline(reconciliator=reconciliator, model=model, transforms=[], horizon=2)
+
+    pipeline.fit(ts=ts)
+    forecast = pipeline.forecast(prediction_interval=True, n_folds=1)
+    np.testing.assert_array_almost_equal(forecast.get_level_dataset(target_level="total").df.values, 10)
+
+
+def test_forecast_reconcile(product_level_constant_hierarchical_ts):
+    ts = product_level_constant_hierarchical_ts
+    reconciliator = BottomUpReconciliator(target_level="total", source_level="product")
+    model = NaiveModel()
+    pipeline = HierarchicalPipeline(reconciliator=reconciliator, model=model, transforms=[], horizon=2)
+
+    pipeline.fit(ts=ts)
+    forecast = pipeline.raw_forecast(prediction_interval=True, n_folds=1)
+    np.testing.assert_array_almost_equal(pipeline.reconciliator.reconcile(ts=forecast).df.values, 10)
+
+
+@pytest.mark.parametrize(
+    "reconciliator",
+    (
+        TopDownReconciliator(target_level="product", source_level="market", period=1, method="AHP"),
+        BottomUpReconciliator(target_level="total", source_level="market"),
+    ),
+)
+def test_width_metric(product_level_constant_hierarchical_ts, reconciliator):
+    ts = product_level_constant_hierarchical_ts
+    model = NaiveModel()
+    pipeline = HierarchicalPipeline(reconciliator=reconciliator, model=model, transforms=[], horizon=1)
+
+    metrics, _, _ = pipeline.backtest(
+        ts=ts,
+        metrics=[Width()],
+        n_folds=2,
+        aggregate_metrics=True,
+        forecast_params={"prediction_interval": True, "n_folds": 1},
+    )
+    np.testing.assert_array_almost_equal(metrics["Width"], 0)
```

Comment:
We actually need to check that this metric can be computed, it is not nessecary to check the exact value

Reply:
This is just for completeness